analysis/bulk-deconvolution/exploratory-notebooks/explore-epic-results.Rmd

```diff
@@ -0,0 +1,401 @@
+    fraction) 
+
+ggplot(classified_samples_df) + 
+  aes(x = sample_diagnosis, y = fraction, fill = epic_celltype) +
+  geom_col() + 
+  scale_fill_manual(values = color_map, na.value = "grey90", name = "cell type") + 
+  theme(axis.text.x = element_text(angle = 30, hjust = 1)) + 
+  facet_wrap(vars(project_diagnosis), scales = "free_x")
+```
+* It does make sense that `CAFs` are dominating osteosarcoma samples (indeed they are prevalent, generally speaking, across a few more than these osteosarcoma samples) since we know they are a major component of osteosarcomas
+  * Some of those cells may actually be tumor cells which `EPIC` misclassified
+* It's worth noting that osteo sample dominated by `CD4 TCells` was also dominated by this cell type for the other `BRef` reference
+
+
+### Number of identified cell types
+
+Let's look at some associated summaries for this sample-specific information.
+
+Of the 7 cell types in `TRef`, how many are represented in each sample?
+For this, we'll make a histogram of the number of cell types identified per sample (regardless of their proportion).
+Since, as it seems, EPIC generally doesn't estimate 0's, we'll instead provide a low threshold value of `1e-4` (artbitrarily chosen as low but not _too_ low!).
+
+```{r}
+celltype_counts_df <- tref_df |>
+  dplyr::filter(!is_other) |>
+  dplyr::group_by(project_diagnosis, sample_id) |>
+  dplyr::summarize(num_celltypes = sum(fraction >= 1e-4))
+```
+
+
+```{r fig.height=4, fig.width=10}
+ggplot(celltype_counts_df) + 
+  aes(x = num_celltypes) + 
+  geom_histogram(center = 0, binwidth = 1) + 
+  scale_x_continuous(breaks = scales::breaks_pretty()) +
+  facet_wrap(vars(project_diagnosis), nrow = 1) + 
+  ggtitle("Number of cell types inferred in individual samples")
+```  
+We can also make an associated summary table:
+
+```{r}
+celltype_counts_df |>
+  dplyr::group_by(project_diagnosis) |>
+  dplyr::summarize(
+    min_celltypes = min(num_celltypes), 
+    median_celltypes = median(num_celltypes), 
+    max_celltypes = max(num_celltypes), 
+  )
+```
+
+Overall, we see that most all samples have at least 5-6 cell types, which is more than half of the possible cell types (there are seven total).
+
+
+Next, in what proportion of samples is each cell type represented?
+Again, this only considers cell type presence and absence at a `1e-4` threshold.
+
+```{r}
+# calculate number of celltypes per sample
+samples_per_celltype_df <- epic_df |>
+  dplyr::filter(!is_other) |>
+  dplyr::mutate(celltype_present = fraction > 1e-4) |>
+  dplyr::group_by(epic_celltype, project_diagnosis) |>
+  dplyr::summarize(n_samples = sum(celltype_present)) 
+
+# convert number to proportion
+samples_per_celltype_df <- epic_df |>
+  dplyr::filter(is_other) |>
+  dplyr::count(project_diagnosis) |>
+  dplyr::right_join(samples_per_celltype_df) |>
+  dplyr::mutate(prop_samples = n_samples / n)
+```
+
+```{r fig.height=4, fig.width=12, message = FALSE}
+ggplot(samples_per_celltype_df) + 
```

Comment:
Is this plot supposed to have both Tref and Bref in the same plot? Cell types from both are listed and I think I would actually separate these out so you show Tref and Bref separately. I also would sort by the proportion.

Reply:
I think that's why you see some numbers that are wrong. For example, CAFs are in almost all samples but that is not represented by this plot.

Reply:
This plot still has some issues. Looking at the stacked bar chart, endothelial cells should be in 100% of samples from SCPCP000001 and SCPCP000002, but the proportion is not even .15?